### ISSUE TITLE
CLion is unable to get compiler information from circle (circle doesn't support -fpch-preprocess and -dD)

### ISSUE THREAD

**Issue by nithrill** [seanbaxter/circle#206]
### TL;DR
circle 205 doesn't support `-fpch-preprocess` or `-dD`, which prevents it from being used as the compiler for CMake projects in CLion.

I believe this to be relevant for circle since I'd expect one of the keys to its adaption being how well it plays with IDEs.

### Details

When using circle as the compiler then any CMake target (e.g.: `add_executable` and `add_library`) generates the following error (as reported within CLion):
```
Cannot get compiler information:
	Compiler exited with error code 1: /usr/bin/circle -xc++ -g -std=c++11 -fpch-preprocess -v -dD -E
	circle: Unknown command line argument '-fpch-preprocess'.  Try: '/usr/bin/circle --help'
	circle: Did you mean '--ftime-report'?
	circle: Unknown command line argument '-dD'.  Try: '/usr/bin/circle --help'
	circle: Did you mean '-D'?
```

sample CMakeLists.txt
```
cmake_minimum_required(VERSION 3.5.1...3.28.1 FATAL_ERROR)

project(test_circle LANGUAGES CXX)

set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_STANDARD 11)

add_executable(test_circle test_circle.cpp)
```

sample test_circle.cpp
```
int main() {}
```

sample CMake command (used by CLion)
```
cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_COMPILER=circle -G "CodeBlocks - Unix Makefiles" -S /tmp/test_circle -B /tmp/test_circle/cmake-build-debug-circle-ubuntu-2404
```

Using CMake on its own doesn't generate any errors.

Searching for the issue on the Internet the best description I could find was in an issue that was raised on the open-watcom issue list back in 2020 (see https://github.com/open-watcom/open-watcom-v2/issues/596).

A possible workaround is to use CLions Custom Compilers feature (released in [CLion 2021.3 EAP](https://blog.jetbrains.com/clion/2021/10/clion-2021-3-eap-custom-compiler/)):
- [Custom Compilers in CLion: A Quick Guide to Using an Arbitrary Compiler in CLion](https://blog.jetbrains.com/clion/2022/09/custom-compilers/)
- [Custom compilers](https://www.jetbrains.com/help/clion/custom-compilers.html)

The issue with using CLions Custom Compilers is that all the data is static (e.g., each time one adds a new `#define` or a new `#include` then the custom section needs to be regenerated).
